addons/pinniped/config-controller/controllers/controllers.go

```diff
@@ -158,17 +146,41 @@ type concierge struct {
 	Audience string `yaml:"audience,omitempty"`
 }
 
-// nolint:funlen,nolintlint // Eh, we can live with a function of this length
-func (c *PinnipedController) reconcileAddonSecret(ctx context.Context, cluster *clusterapiv1beta1.Cluster, pinnipedInfoCM *corev1.ConfigMap) error {
-	log := c.Log.WithValues(constants.NamespaceLogKey, cluster.Namespace, constants.NameLogKey, cluster.Name)
-	// check if cluster is scheduled for deletion, if so, delete addon secret on mgmt cluster
-	if !cluster.GetDeletionTimestamp().IsZero() {
-		log.V(1).Info("cluster is getting deleted, deleting addon secret")
-		if err := c.reconcileClusterDelete(ctx, client.ObjectKeyFromObject(cluster)); err != nil {
-			return fmt.Errorf("failed to reconcile cluster delete: %w", err)
+// TODO: simplify this function
+// nolint:funlen,nolintlint,gocyclo // Eh, we can live with a function of this length
+func (c *PinnipedController) reconcileSecret(ctx context.Context, secret *corev1.Secret, pinnipedInfoCM *corev1.ConfigMap) error {
+	secretName := secret.Name
+	secretNamespace := secret.Namespace
+	log := c.Log.WithValues(constants.SecretNamespaceLogKey, secretNamespace, constants.SecretNameLogKey, secretName)
+	// check if secret is scheduled for deletion, if so, skip reconcile
+	if !secret.GetDeletionTimestamp().IsZero() {
+		log.V(1).Info("secret is getting deleted, skipping reconcile")
+		return nil
+	}
+	clusterName, err := utils.GetClusterNameFromSecret(secret)
+	if err != nil {
+		log.Error(err, "could not get cluster name from secret")
```

Comment:
shall we specify the secret in logs again?

Reply:
This line above adds that information to all the logs:
```
log := c.Log.WithValues(constants.SecretNamespaceLogKey, secretNamespace, constants.SecretNameLogKey, secretName)
```

here is more information about this convention: https://github.com/kubernetes-sigs/controller-runtime/blob/9ee63fc65a9720568f74df9901b16883621636b4/TMP-LOGGING.md